### PR TITLE
feat(OMN-11069): Wave 1 ModelOverlayFile + ModelOverlayResolutionManifest (OMN-11070)

### PR DESCRIPTION
**Pull request by jonahgabriel**
## Summary

- Add `ModelOverlayFile` — frozen Pydantic schema for overlay YAML files with `overlay_version` (ModelSemVer), `scope` (EnumOverlayScope), `transports`/`secrets`/`services`/`llm` sections, `content_hash()`, `all_env_pairs()` with conflict detection, and `redacted_dump()` masking sensitive keys
- Add `ModelOverlayResolutionManifest` — frozen evidence artifact for runtime overlay resolution with `stable_identity_hash()` that excludes `timestamp` and `runtime_version` for deterministic cross-boot comparison
- Part of OMN-11069: contract-overlay configuration to replace .env files
- Implements OMN-11070: Wave 1 core models

## Ticket

OMN-11069 OMN-11070

Evidence-Source: OCC#1045
Evidence-Ticket: OMN-11070

## Test plan

- [x] 18 unit tests covering validation, hash determinism, collision detection, secret redaction
- [x] `uv run mypy src/omnibase_core/models/overlay/ --strict` — no issues
- [x] `pre-commit run --all-files` — all hooks pass

## Summary by CodeRabbit

* **New Features**
  * Added overlay configuration model with version validation, environment-variable aggregation, secret redaction, and deterministic content hashing.
  * Added overlay resolution manifest capturing resolved state with a stable identity hash.

* **Tests**
  * Added comprehensive unit tests covering validation, immutability, hashing stability, env-pair aggregation, redaction, and manifest identity behavior.

[Review Change Stack](https://app.coderabbit.ai/change-stack/OmniNode-ai/omnibase_core/pull/1082?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by jonahgabriel** [comment]
Addressing CodeRabbit review findings (threads 3250883162, 3250883173, 3250883177, 3250883179):

**Thread 1 & 2 (docstrings):** Fixed in commit e3685558 — added class docstring to `ModelOverlayFile` and method docstrings to `content_hash()`, `all_env_pairs()`, and `redacted_dump()` describing non-obvious behavior.

**Thread 3 (error-ok vs boundary-ok):** The comment marker `# error-ok:` is correct here. The repo's `check-error-raising` pre-commit hook (scripts/validation/check_error_raising.py) specifically requires `# error-ok:` annotation on bare `raise ValueError` calls — this is mechanically enforced by CI. The CLAUDE.md's `# boundary-ok:` marker is for a different category of annotation (graceful degradation). Using `# boundary-ok:` here would break the `ONEX Error Raising Validation` pre-commit hook.

**Thread 4 (type: ignore annotation):** Fixed in commit e3685558 — added `# NOTE(OMN-11070):` comment explaining why the ignore is needed (mypy correctly flags assignment to a frozen model field; the ignore is intentional to test runtime enforcement).

File: tests/unit/models/overlay/test_model_overlay_resolution_manifest.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            "contract_requirements_hash": "sha256:def456",
+            "resolved_config_hash": "sha256:789abc",
+            "resolved_transports": ("database", "kafka"),
+            "required_transports": ("database", "kafka", "valkey"),
+            "runtime_version": "0.35.0",
+            "timestamp": datetime.now(tz=UTC),
+            "config_source": "overlay",
+        }
+        defaults.update(overrides)
+        return ModelOverlayResolutionManifest(**defaults)  # type: ignore[arg-type]
```
[the remark this answers is not in the document]
Comment (jonahgabriel) [reply tests/unit/models/overlay/test_model_overlay_resolution_manifest.py:31]: The fix was applied in commit e3685558: line 40 now has `# NOTE(OMN-11070): mypy flags assignment to frozen model field; ignore is intentional to test runtime enforcement` immediately before the `type: ignore[misc]` annotation. This satisfies the guideline requiring a NOTE comment with the ticket reference explaining why the ignore is necessary.

File: src/omnibase_core/models/overlay/model_overlay_file.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        sources.append(("secrets", self.secrets))
+        for section_name, svc_vals in self.services.items():
+            sources.append((f"services.{section_name}", svc_vals))
+        for section_name, llm_vals in self.llm.items():
+            sources.append((f"llm.{section_name}", llm_vals))
+
+        for source_label, kv in sources:
+            for key, value in kv.items():
+                if key in pairs and pairs[key] != value:
+                    raise ValueError(  # error-ok: boundary ValueError for duplicate key in public API
```
[the remark this answers is not in the document]
Comment (jonahgabriel) [reply src/omnibase_core/models/overlay/model_overlay_file.py:90]: This codebase uses `# error-ok:` (not `# boundary-ok:`) as the annotation for intentional `raise ValueError(` at public API boundaries. The `check-error-raising` pre-commit hook enforces this convention: it flags bare `raise ValueError(` without an `# error-ok:` annotation and is wired as a required CI check across this repo. Using `# boundary-ok:` here would fail the pre-commit hook. The `# error-ok: boundary ValueError for duplicate key in public API` annotation follows the exact convention used throughout `omnibase_core` (e.g., `model_overlay_file.py:78`). This is correct behavior for this codebase.

File: tests/unit/models/overlay/test_model_overlay_resolution_manifest.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonahgabriel) [reply tests/unit/models/overlay/test_model_overlay_resolution_manifest.py:31]: Applied in commit df2f8a17: added `# NOTE(OMN-11070): mypy cannot statically verify the dynamically-built defaults dict matches the constructor signature` on the line before `return ModelOverlayResolutionManifest(**defaults)  # type: ignore[arg-type]` in `_make_manifest`. Both `type: ignore` annotations now have NOTE comments.
